compiler/rustc_query_system/src/dep_graph/graph.rs

```diff
@@ -115,9 +125,68 @@ impl<K: DepKind> DepGraph<K> {
         encoder: FileEncoder,
         record_graph: bool,
         record_stats: bool,
+        shared: bool,
```

Comment:
If the wrong bool is passed here, this ends up introducing UB, right? That means that either this needs to not be a bool (but rather some kind of token type that's only possible to create in single threaded compilers, or this function needs to be unsafe (which is terrible).

Reply:
That makes sense. I think a static `OnceLock` is an option